.travis.yml

```diff
@@ -22,10 +25,12 @@ install:
   - python setup.py install
   - python -m nltk.downloader wordnet
 
+env:
+  - TEST_SUITE=test/unit-test
```

Comment:
I only read about travis build matrix. I agree that we got our necessary parallelism.
From a semantic perspective, this is really awkward though. Can you provide a reference to travis documentation that suggests this kind of coding idiom?

Reply:
another side effect of the suggested arrangement is that the code coverage report is an aggregation of unit and functional tests. We want to keep both the reports separate.

Reply:
https://docs.travis-ci.com/user/speeding-up-the-build/

Reply:
No, sir. The codecoverage report is now separate.  Earlier it was an aggregation. This is evident in the output of `coverage report -m`

Reply:
The `coverage report -m` output for both unit and functional tests on a stand alone basis is correct. But both these reports get pushed to codecov, which does its own aggregation (that's the only way to explain the increase in code coverage with no change in code base).
We are also sending both the reports to codeclimate.

The codecoverage arrangement we are planning is as follows.
1. unit tests ---> codecov
1. functional tests ---> coveralls
1. integration tests ---> codeclimate

Reply:
@bhuvangupta008 Since we are going to start the integration tests after unit tests, it is better to add the integration tests to build matrix as well. At present, you can create an empty integration tests directory and run the unit test on the same. That would be a skeleton which we can modify later on.